### PR TITLE
KAFKA-18226: Disable CustomQuotaCallbackTest and remove `isKRaftTest`

### PR DESCRIPTION
**Pull request by m1a2st**
Jira: https://issues.apache.org/jira/browse/KAFKA-18226

We change all tests to kraft mode, thus we didn't need to check which test is zk, we can remvoe check zk condition 

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala [hunk 1]
```diff
@@ -55,7 +55,7 @@ abstract class BaseProducerSendTest extends KafkaServerTestHarness {
     overridingProps.put(ServerLogConfigs.NUM_PARTITIONS_CONFIG, 4.toString)
     TestUtils.createBrokerConfigs(
       numServers,
-      zkConnectOrNull,
+      null,
```
Comment (mimaison) [inline core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala:58]: Could we even remove the argument from `createBrokerConfigs()`?

Comment (m1a2st) [reply core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala:58]: The `createBrokerConfigs()` will modify about 40 files, I will address it on another PR and open another Jira to trace it.

Update: I open Jira: [KAFKA-18261](https://issues.apache.org/jira/browse/KAFKA-18261)

Comment (m1a2st) [reply core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala:58]: Some tests are still using not `null` value on this method.
https://github.com/apache/kafka/blob/ed10fc63a9fd3032a529f7e10a0f863ee71b710a/core/src/test/scala/unit/kafka/server/ReplicaAlterLogDirsThreadTest.scala#L208

Comment (mimaison) [reply core/src/test/scala/integration/kafka/api/BaseProducerSendTest.scala:58]: That makes sense, then we can do this in another PR. Thanks

File: core/src/test/scala/unit/kafka/server/BaseRequestTest.scala [hunk 2]
```diff
@@ -72,13 +72,7 @@ abstract class BaseRequestTest extends IntegrationTestHarness {
...
    * controller. For Legacy clusters that is the controller broker.
    */
-  def adminSocketServer: SocketServer = {
-    if (isKRaftTest()) {
-      anySocketServer
-    } else {
-      controllerSocketServer
-    }
-  }
+  def adminSocketServer: SocketServer = anySocketServer
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/server/BaseRequestTest.scala:74]: Can we adjust the comment?

File: core/src/test/scala/integration/kafka/api/CustomQuotaCallbackTest.scala [hunk 3]
```diff
@@ -42,6 +41,7 @@ import java.{lang, util}
 import scala.collection.mutable.ArrayBuffer
 import scala.jdk.CollectionConverters._
 
+@Disabled
```
Comment (mimaison) [inline core/src/test/scala/integration/kafka/api/CustomQuotaCallbackTest.scala]: -> `@Disabled("KAFKA-18213")`

File: core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala [hunk 4]
```diff
@@ -102,20 +98,17 @@ abstract class IntegrationTestHarness extends KafkaServerTestHarness {
...
-      props.foreach { config =>
-        // Add a security protocol for the controller endpoints, if one is not already set.
-        val securityPairs = config.getProperty(SocketServerConfigs.LISTENER_SECURITY_PROTOCOL_MAP_CONFIG, "").split(",")
-        val toAdd = config.getProperty(KRaftConfigs.CONTROLLER_LISTENER_NAMES_CONFIG, "").split(",").filter(
-          e => !securityPairs.exists(_.startsWith(s"$e:")))
-        if (toAdd.nonEmpty) {
-          config.setProperty(SocketServerConfigs.LISTENER_SECURITY_PROTOCOL_MAP_CONFIG, (securityPairs ++
-            toAdd.map(e => s"$e:${controllerListenerSecurityProtocol.toString}")).mkString(","))
-        }
+  private def insertControllerListenersIfNeeded(props: Seq[Properties]): Unit =
```
Comment (mimaison) [inline core/src/test/scala/integration/kafka/api/IntegrationTestHarness.scala]: Can we keep the curly brackets `{` and `}` around the method?
```
private def insertControllerListenersIfNeeded(props: Seq[Properties]): Unit = {
...
}
```

File: core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala [hunk 5]
```diff
@@ -482,18 +365,6 @@ abstract class QuorumTestHarness extends Logging {
...
-    }
-    val anotherZkClient = new ZooKeeper(zkConnect, 1000, dummyWatcher,
-      zooKeeper.getSessionId,
-      zooKeeper.getSessionPasswd)
-    assertNull(anotherZkClient.exists("/nonexistent", false)) // Make sure new client works
-    anotherZkClient
-  }
 }
 
 object QuorumTestHarness {
```
Comment (mimaison) [inline core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala:370]: Can we also remove the `ZkClientEventThreadSuffix` field from this object? Or is it still used?

File: core/src/test/scala/unit/kafka/integration/KafkaServerTestHarness.scala [hunk 6]
```diff
@@ -176,30 +165,18 @@ abstract class KafkaServerTestHarness extends QuorumTestHarness {
...
-          controllers = controllerServers,
-          numPartitions = numPartitions,
-          replicationFactor = replicationFactor,
-          topicConfig = topicConfig
-        )
-      }
-    } else {
-      TestUtils.createTopic(
-        zkClient = zkClient,
+  ): scala.collection.immutable.Map[Int, Int] =
```
Comment (mimaison) [inline core/src/test/scala/unit/kafka/integration/KafkaServerTestHarness.scala]: Let's keep the brackets around the method here too.

Comment (mimaison) [reply core/src/test/scala/unit/kafka/integration/KafkaServerTestHarness.scala]: Same for the other methods below

**Review by mimaison — COMMENTED** [review]
Thanks for the PR. I left a few comments.

File: core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala (hunk 5 above)
Comment (m1a2st) [reply core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala:370]: It still use on this, but I think we can remove it https://github.com/apache/kafka/blob/220c57852100e895b460e56d1132a3c9239aaa59/core/src/test/scala/integration/kafka/server/QuorumTestHarness.scala#L530

**Review by mimaison — APPROVED** [review]
Thanks for the PR, LGTM

**Pull request merged**